include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -78,6 +80,148 @@ __find_start_point(const _Rng1& __rng1, const _Rng2& __rng2, const _Index __i_el
     }
 }
 
+//Searching for an intersection of a merge matrix (n1, n2) diagonal with the Merge Path to define sub-ranges
+//to serial merge. For example, a merge matrix for [0,1,1,2,3] and [0,0,2,3] is shown below:
+//     0   1  1  2   3
+//    ------------------
+//   |--->
+// 0 | 0 | 1  1  1   1
+//   |   |
+// 0 | 0 | 1  1  1   1
+//   |   ---------->
+// 2 | 0   0  0  0 | 1
+//   |             ---->
+// 3 | 0   0  0  0   0 |
+template <typename _Rng1, typename _Rng2, typename _Index, typename _Compare>
+_split_point_t<_Index>
+__find_start_point_in(const _Rng1& __rng1, const _Index __rng1_from, _Index __rng1_to, const _Rng2& __rng2,
+                      const _Index __rng2_from, _Index __rng2_to, const _Index __i_elem, _Compare __comp)
```

Comment:
If a "full" unification of the implementation is not feasible, I would at least try to unify `__find_start_point` and `__find_start_point_in` into a single function. In theory it should be possible, because `const _Rng1& __rng1, const _Index __rng1_from, _Index __rng1_to` together really define a subrange of `__rng1`; so perhaps we could create it on the caller side and pass to the function as the range to process.

Reply:
There is one moment.
My checks shows me that `__find_start_point_in` is more effective on middle and large data sizes
and `__find_start_point` is more effective on smaller data sizes.
That was the reason why I save old implementation.
But technically I can use `__find_start_point_in` in all cases: for smaller data sizes some performance degradation will have place.

Reply:
So I propose to save both `__find_start_point` and `__find_start_point_in` functions.
The justification - to have the same performance on the sizes less then 4 Mb.

Reply:
> That was the reason why I save old implementation.
> But technically I can use `__find_start_point_in` in all cases: for smaller data sizes

In that case should be a comment in the code why we have two implementations.

Reply:
I never seen these explanations in out code.